test/js/node/http/node-http.test.ts

```diff
@@ -715,6 +715,114 @@ describe("node:http", () => {
         req.end();
       });
     });
+
+    it("support custom tls args", done => {
```

Comment:
only other thing i can think of here is what to do about redirects

but we can merge this